src/tests/issues.targets

```diff
@@ -1221,6 +1221,9 @@
         <ExcludeList Include="$(XunitTestBinBase)/Interop/StringMarshalling/BSTR/BSTRTest/**">
             <Issue>Crashes during LLVM AOT compilation.</Issue>
         </ExcludeList>
+        <ExcludeList Include="$(XunitTestBinBase)/Interop/MarshalAPI/FunctionPointer/FunctionPtrTest/*">
+            <Issue>Generic calli support not implemented yet in mono.</Issue>
```

Comment:
How hard is it to implement generic unmanaged calli support in Mono?

cc @vargaz @AlekseyTs

Reply:
Based on some of my conversations with @lambdageek I think generics support at some of these interop margins are a bit tricky in mono.

Reply:
Hmm, that's unfortunate.

We try to maintain feature parity between all runtimes. We need to figure out what we are going to do about Mono support before enabling it for CoreCLR.

Reply:
The example from the issue descriptions works in net8.0 Mono already with JIT and interpreter.  I haven't tried the examples from the rest of the added test.

--- 

The thing that Aaron is talking about is that mono doesn't have great support for creating wrapper methods that are themselves generic (ie: they have generic method params).  We would need that for the generic `UnsafeAccessor` work - the issue there is that we need to generate a generic method.  Likewise if it was legal to do something like this:

```csharp

void *pGenericFn = &SomeGenericUnsafeCallersOnlyMethod;

[UnsafeCallersOnly]
unsafe void SomeGenericUnsafeCallersOnlyMethod<T>(T *) where T : unmanaged { ... }
```

we would need to generate a generic wrapper.

---

For unmanaged function pointers we create wrappers around the `&Func` - `ldftn SomeUnmanagedCallersOnlyMethod` actually returns the address of the wrapper that has the native, not managed, calling convention.

At the `calli`, we don't need to do anything special almost all the time.

The one place where we might get into trouble is if we need to do a call out using generic sharing, or gsharedvt (universal generic sharing):  That is `Foo<int>.Call<int>(...)` from the issue description, but compiled for ios, for example.  I haven't tried it.

Reply:
The failures are probably going to be with generic sharing and aot. The JIT should work fine.

Reply:
> Hmm, that's unfortunate.
> 
> We try to maintain feature parity between all runtimes. We need to figure out what we are going to do about Mono support before enabling it for CoreCLR.

The tests were actually passing on mono, except mono-llvm-aot failed to build the negative tests (RunInvalidGenericFunctionPointerTest).

Reply:
Could you please factor out the negative cases into separate test and only disable those for Mono? 

Handling of negative cases by llvm-aot should not be blocking.

Reply:
Seems that even if we removed negative cases, mono-aot-llvm still fail to compile (but with a different error message).